### PR TITLE
ci/docs: use python 3.9

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

alow us to use higher Python versions transparently

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca

### CODE REVIEW

**Comment by Borda** [comment]
NOTE: some past docs build failed silently:
- https://github.com/Lightning-AI/lightning/actions/runs/5510378761
- https://github.com/Lightning-AI/lightning/actions/runs/5650063485

File: .github/workflows/docs-build.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
     tags: ["*"]
   # use this event type to share secrets with forks.
   # it's important that the PR head SHA is checked out to run the changes
-  pull_request_target:
+  pull_request:  # fixme: pull_request_target
```
Comment (Borda) [inline .github/workflows/docs-build.yml:10]:
```suggestion
  pull_request_target:
```

File: .github/workflows/ci-dockers.yml [hunk 2]
```diff
@@ -123,6 +123,7 @@ jobs:
         timeout-minutes: 55
 
   build-docs:
+    # todo: this may not be needed
```
Comment (justusschock) [inline .github/workflows/ci-dockers.yml:126]: are you planning to resolve this todo in this pr?

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-dockers.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-dockers.yml:126]: dropped

**Pull request merged**